utils/pattern/src/implementations.rs

```diff
@@ -0,0 +1,28 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{Pattern, SinglePlaceholder, SinglePlaceholderPattern};
+
+use zerovec::__zerovec_internal_reexport::boxed::Box;
+use zerovec::{maps::ZeroMapKV, ule::VarULE, VarZeroSlice, VarZeroVec, ZeroVecError};
+
+impl<'a> ZeroMapKV<'a> for Pattern<SinglePlaceholder, str> {
+    type Container = VarZeroVec<'a, Pattern<SinglePlaceholder, str>>;
+    type Slice = VarZeroSlice<Pattern<SinglePlaceholder, str>>;
+    type GetType = Pattern<SinglePlaceholder, str>;
+    type OwnedType = Box<Pattern<SinglePlaceholder, str>>;
```

Comment:
should this be `Pattern<SinglePlaceholder, String>`?

Reply:
Yes if it's easy, but not super important

Reply:
it's just another transmute

Reply:
why it should be `Pattern<SinglePlaceholder, String>` , if we are using `Pattern<SinglePlaceholder, str>` in the datagen ?

Reply:
the implementation will be : https://github.com/unicode-org/icu4x/pull/5084